### DISCUSSION TITLE
I can't get pooling connection with supabase-js to work on a Vercel serverless function in NodeJS

### DISCUSSION THREAD

**Discussion by Evertt** [supabase/supabase#13003] [Questions]
I don't know what I'm doing wrong.

I'm creating the connection like so:

```typescript
const supabase = createClient(
  `https://db.thisisredactedobviously.supabase.co:6543/postgres?pgbouncer=true`,
  SUPABASE_KEY,
)
```

But then if I run any query, like this one for example:

```typescript
async set(key: string, session: Session) {
  const { error } = await supabase
    .from('telegraf-sessions')
    .upsert({ key, session })
    .single()

    console.log("Stored a session in supabase:", { key, session, error })

    return error ? { error } : true
}
```

Where `session` is a json object, but the column in the database table also is of type `jsonb`. I always get an error, saying:
```
{
    message: 'FetchError: fetch failed',
    details: '',
    hint: '',
    code: ''
  }
```

Why? What am I doing wrong? I turned off `RLS`, just to make sure that that wasn't the problem. But then what is the problem?

**Comment by GaryAustin1** [comment]
createClient does not work through the postgres database ports.  You use the instance url only.   The database API has its own pooler built in.

**Reply by Evertt** [reply]
I'm sorry, I don't fully understand what you are saying.

I understand you're saying that I can only use `createClient` like so:

```js
const supabase = createClient(
  "https://db.thisisredactedobviously.supabase.co",
  SUPABASE_KEY,
)
```

But then when you say:

> The database API has its own pooler built in.

I'm not sure if you mean that the client I just created will just automatically use the pooler, since the pooler is enabled in my supabase database settings. Or if you are saying that I should use something completely different than `createClient()`. Can you please clear up my confusion? 😇

**Reply by GaryAustin1** [reply]
PostgREST the database REST API Supabase uses has a it’s own pooler built in.  So you don’t need to worry about using the database pooler port when using Supabase-js which calls the REST API for database calls.
